File: lc/409/longest-palindrome.c
int longestPalindrome(char *s) {
   size_t dict[255] = {};
   while (*s != 0) {
      dict[*s++]++;
   }

   size_t sum = 0;
   size_t middle = 0;

   for (size_t i = 0; i < 255; i++) {
      if (dict[i] % 2 == 1) {
         middle = 1;
      }

      sum += 2 * (dict[i] / 2);
   }

   return sum + middle;
}

File: lc/409/longest-palindrome.ts
function longestPalindrome(s: string): number {
   const chars: {[char: string]: number} = {};
   for (const c of s) {
      if (c in chars) {
         chars[c]++;
      } else {
         chars[c] = 1;
      }
   }

   let sum = 0;
   let middle = 0;
   for (const count of Object.values(chars)) {
      if (count % 2 == 1) {
         middle = 1;
      }
      const pair = count / 2 | 0;
      sum += pair * 2;
   }

   return sum + middle;
}
